spec/controllers/comments/comments_controller_spec.rb

```diff
@@ -491,4 +491,15 @@
       before { work.update!(collection_names: create(:unrevealed_collection).name) }
     end
   end
+
+  describe "POST #create" do
+    context "when commentable has posted that is falsy" do
+      it "does not allow comment creation" do
+        work = create(:work, posted: false)
+        fake_login_known_user(work.users.first)
+        post :create, params: { work_id: work.id, comment: { comment_content: "Test comment" } }
+        expect(flash[:error]).to eq("Sorry, you can't add or edit comments on an unpublished work.")
+      end
+    end
+  end
```

Comment:
This should be updated to use the `no one can add or edit comments` shared example (used in the unrevealed and hidden tests just above)